x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -335,12 +343,13 @@ public void close() {
         }
     }
 
-    private static class RestoreFileInputStream extends InputStream {
+    private class RestoreFileInputStream extends InputStream {
```

Comment:
keep it static and add CombinedRateLimiter as constructor parameter (same for RestoreSession)?